src/test/java/com/uber/cadence/workflow/interceptors/SignalWorkflowInterceptor.java

```diff
@@ -0,0 +1,127 @@
+/*
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.uber.cadence.workflow.interceptors;
+
+import com.uber.cadence.WorkflowExecution;
+import com.uber.cadence.activity.ActivityOptions;
+import com.uber.cadence.workflow.*;
+
+import java.lang.reflect.Type;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Random;
+import java.util.UUID;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class SignalWorkflowInterceptor implements WorkflowInterceptor {
```

Comment:
I don't think this class and the factory are generally useful. Let's move them under the test.

Reply:
The ability to override signals is potentially useful. The factory is test specific, it overrides the signal in a very specific way to the test so I moved it there.